### PR TITLE
Partner Portal: Add a Terms of Service consent gate to access the portal

### PR DESCRIPTION
**Pull request by atanas-dev**
Context: p1616184557016100-slack-CQN0YDYM7

#### Changes proposed in this Pull Request

* Add a Terms of Service consent gate to access the Partner Portal.

#### Testing instructions

* If you do not have a partner key, please contact Infinity for one or you will be unable to view test this PR.
* Check out this PR locally, then run `yarn && yarn start-jetpack-cloud`.

Case 1: You don't have a partner or at least 1 active partner key.
* Visit http://jetpack.cloud.localhost:3000/partner-portal
* You should be redirected to /partner-portal/partner which should show an error message like so:
Screenshot 2021-03-23 at 18 04 45

Case 2: You have an active partner key but you have not accepted the new ToS.
* Visit http://jetpack.cloud.localhost:3000/partner-portal
* You should be redirected to /partner-portal/terms-of-service:
Screenshot 2021-04-13 at 15 55 19
* Accepting the terms should redirect you to key selection if you have multiple keys or to your license listing if you only have a single key.

Case 3: You have an active partner key and you have already accepted the new ToS once.
* Visit http://jetpack.cloud.localhost:3000/partner-portal
* You should be redirected to key selection if you have multiple keys or to your license listing if you only have a single key, skipping the ToS step.

### CODE REVIEW

File: client/jetpack-cloud/sections/partner-portal/primary/terms-of-service-consent/index.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+export default function TermsOfServiceConsent(): ReactElement | null {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const consent = useMutation( mutationConsent, {
+		onSuccess: ( partner ) => {
+			dispatch( receivePartner( formatApiPartner( partner ) ) );
+		},
+		onError: () => {
+			dispatch(
+				errorNotice( translate( 'We were unable to update your partner account details.' ) )
```
Comment (a8ci18n) [inline client/jetpack-cloud/sections/partner-portal/primary/terms-of-service-consent/index.tsx:44]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'We were unable to retrieve your partner account details.' )` *ES Score: 8*

**Comment by IanRamosC** [comment]
The functionality tests well. Besides the blockers, I would only ping @keoshi to check the UI before releasing it.

**Comment by keoshi** [comment]
This looks good to me so far. Would like to see what the rest of the UI looks like... Are we showing menus when the TOS is display? I don't think we should show anything until the TOS are accepted, right?

I can also spin the PR later today and test it for myself, but that would be my only concern for now.

Finally, just said this to @atanas-dev but will say it again — apologies for not having replied here sooner. I got the email ping but it's been sitting on my todo list because I'm extra busy lately.

**Review by IanRamosC — APPROVED** [review]
Looks good!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/5702430

Thank you @atanas-dev for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/5702430).

**Pull request merged**